rotkehlchen/chain/evm/decoding/rainbow/decoder.py

```diff
@@ -0,0 +1,417 @@
+import logging
+from collections.abc import Callable
+from typing import TYPE_CHECKING, Optional
+
+from rotkehlchen.assets.asset import CryptoAsset
+from rotkehlchen.chain.ethereum.utils import asset_normalized_value, asset_raw_value
+from rotkehlchen.chain.evm.decoding.constants import ERC20_OR_ERC721_TRANSFER
+from rotkehlchen.chain.evm.decoding.interfaces import DecoderInterface
+from rotkehlchen.chain.evm.decoding.structures import (
+    FAILED_ENRICHMENT_OUTPUT,
+    EnricherContext,
+    TransferEnrichmentOutput,
+)
+from rotkehlchen.chain.evm.decoding.types import CounterpartyDetails
+from rotkehlchen.chain.evm.decoding.utils import maybe_reshuffle_events
+from rotkehlchen.chain.evm.decoding.weth.decoder import WETH_DEPOSIT_TOPIC, WETH_WITHDRAW_TOPIC
+from rotkehlchen.chain.evm.structures import EvmTxReceiptLog
+from rotkehlchen.errors.asset import UnsupportedAsset, WrongAssetType
+from rotkehlchen.history.events.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.types import ChecksumEvmAddress, EvmTransaction, EVMTxHash, Timestamp
+from rotkehlchen.utils.misc import bytes_to_address
+
+from .constants import CPT_RAINBOW_SWAPS
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.evm.decoding.base import BaseDecoderTools
+    from rotkehlchen.chain.evm.node_inquirer import EvmNodeInquirer
+    from rotkehlchen.history.events.structures.evm_event import EvmEvent
+    from rotkehlchen.user_messages import MessagesAggregator
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class RainbowCommonDecoder(DecoderInterface):
+
+    def __init__(
+            self,
+            evm_inquirer: 'EvmNodeInquirer',
+            base_tools: 'BaseDecoderTools',
+            msg_aggregator: 'MessagesAggregator',
+            router_address: ChecksumEvmAddress,
+    ) -> None:
+        super().__init__(evm_inquirer, base_tools, msg_aggregator)
+        self.router_address = router_address
+
+    def _enricher_event(
+            self,
+            event: 'EvmEvent',
+            sender: ChecksumEvmAddress,
+    ) -> None:
+        """Set the event to rainbow event
+        """
```

Comment:
Rename the function to `_set_event_properties` but also keeps docstrings in one line when possible 

```python
"""Set properties of the event to a rainbow swap considering the event direction"""